### PR TITLE
Document experimental JSON modules and es-module-specifier-resolution

### PR DESCRIPTION
**Pull request by MylesBorins**
Not currently documented, now documented 🎉

Would like to land asap so please review

### CODE REVIEW

File: doc/api/cli.md [hunk 1]
```diff
@@ -134,6 +134,25 @@ added: v6.0.0
...
+<!-- YAML
+added: REPLACEME
+-->
+
+To be used in conjunction with `--experimental modules`. Sets the resolution
+algorithm for resolving specifiers. Valid options are `explicit` and `node`.
+
+The default is `explicit`, which requires you to provide the full path to a module.
+The `node` mode will enable support for optional file extensions and the ability
+to import a directory that has an index file.
```
Comment (GeoffreyBooth) [inline doc/api/cli.md]: Can we add an example? “For example, `--es-module-specifier-resolution=node` allows `import './file'` whereas `explicit` requires `import './file.js'`“ or something like that.

File: doc/api/cli.md [hunk 2]
```diff
@@ -134,6 +134,25 @@ added: v6.0.0
...
+
+### `--experimental-json-modules`
+<!-- YAML
+added: REPLACEME
+-->
+
+Enable experimental JSON support for ES modules.
```
Comment (GeoffreyBooth) [inline doc/api/cli.md]: Can we mention that this is for `import` of JSON?

File: doc/api/esm.md [hunk 3]
```diff
@@ -390,6 +390,23 @@ fs.readFileSync = () => Buffer.from('Hello, ESM');
...
+being standardized, and are experimentally supported by including the
+additional flag `--experimental-json-modules` when running Node.js.
+
+When the `--experimental-json-modules` flag is included both the
+`commonjs` and `modules` mode will use the new experimental JSON
+loader. The imported JSON only exposes a `default`, there is no
+support for named exports. A cache entry is created in the CommonJS
+cache, to avoid duplication. The same object will be returned in
+CommonJS if the JSON module has already been imported from the
+same path.
```
Comment (GeoffreyBooth) [inline doc/api/esm.md]: Should we include an example?

```js
import packageConfig from './package.json';
```

File: doc/api/esm.md [hunk 4]
```diff
@@ -659,6 +676,18 @@ READ_PACKAGE_JSON(_packageURL_)
 
 </details>
 
+### Customizing ESM specifier resolution algorithm
+
+The current specifier resolution does not support all default behavior of
+the Common.js loader. One of the behavior differences is automatic resolution
```
Comment (guybedford) [inline doc/api/esm.md]: CommonJS

**Comment by MylesBorins** [comment]
landed in 9301a063d9

**Pull request closed without merging**